src/libraries/System.Text.Json/Common/JsonSeparatorNamingPolicy.cs

```diff
@@ -0,0 +1,157 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal abstract class JsonSeparatorNamingPolicy : JsonNamingPolicy
+    {
+        private readonly bool _lowercase;
+        private readonly char _separator;
+
+        internal JsonSeparatorNamingPolicy(bool lowercase, char separator) =>
+            (_lowercase, _separator) = (lowercase, separator);
+
+        public sealed override string ConvertName(string name)
+        {
+            // Rented buffer 20% longer that the input.
+            int bufferLength = (12 * name.Length) / 10;
+            char[]? buffer = bufferLength > JsonConstants.StackallocCharThreshold
+                ? ArrayPool<char>.Shared.Rent(bufferLength)
+                : null;
+
+            int resultLength = 0;
+            Span<char> result = buffer is null
```

Comment:
Then `result` can be renamed to `buffer`.
```suggestion
            Span<char> buffer = rentedBuffer is null
```

Reply:
Is it really better than `result` which explicitly tells that that span contains data which is a resulting string content?